### PR TITLE
MGMT-4447 SNO - if machine cidr was not set, route53 fails

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/network/machine_network_cidr.go [hunk 1]
```diff
@@ -255,6 +255,25 @@ func GetMachineCidrForUserManagedNetwork(cluster *common.Cluster, log logrus.Fie
 	return ""
 }
 
+func GetIpForSingleNodeInstallation(cluster *common.Cluster, log logrus.FieldLogger) (string, error) {
+	bootstrap := common.GetBootstrapHost(cluster)
+	if bootstrap == nil {
+		return "", errors.Errorf("no bootstrap host were found in cluster")
+	}
+	cidr := GetMachineCidrForUserManagedNetwork(cluster, log)
+	cluster.MachineNetworkCidr = cidr
```
Comment (eranco74) [inline internal/network/machine_network_cidr.go:264]: Why is this here?
Seems irrelevant to what this function says it does

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -3898,7 +3898,13 @@ func (b *bareMetalInventory) changeDNSRecordSets(ctx context.Context, cluster co
 		apiVip := cluster.APIVip
 		ingressVip := cluster.IngressVip
 		if common.IsSingleNodeCluster(&cluster) {
-			apiVip, _ = network.GetMachineCIDRIP(common.GetBootstrapHost(&cluster), &cluster)
+			apiVip, err = network.GetIpForSingleNodeInstallation(&cluster, log)
+			if err != nil {
+				log.WithError(err).Errorf("failed to update DNS record: (%s, %s)",
```
Comment (eranco74) [inline internal/bminventory/inventory.go:3903]: Misleading log line,
it failed to get the IP not to set the api-int DNS

File: internal/network/machine_network_cidr.go (hunk 1 above)
Comment (tsorya) [reply internal/network/machine_network_cidr.go:264]: this is the way to provide machineCidr to the next one

File: internal/network/machine_network_cidr.go [hunk 3]
```diff
@@ -255,6 +255,25 @@ func GetMachineCidrForUserManagedNetwork(cluster *common.Cluster, log logrus.Fie
...
+	hostIp, err := getMachineCIDRObj(bootstrap, cluster, "ip")
```
Comment (eranco74) [inline internal/network/machine_network_cidr.go:265]: Consider passing machineCidr instead of passing the cluster.

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -3898,7 +3898,13 @@ func (b *bareMetalInventory) changeDNSRecordSets(ctx context.Context, cluster co
 		apiVip := cluster.APIVip
 		ingressVip := cluster.IngressVip
 		if common.IsSingleNodeCluster(&cluster) {
-			apiVip, _ = network.GetMachineCIDRIP(common.GetBootstrapHost(&cluster), &cluster)
+			apiVip, err = network.GetIpForSingleNodeInstallation(&cluster, log)
+			if err != nil {
+				log.WithError(err).Errorf("failed to find ip for DNS record: (%s, %s)",
+					domain.APIINTDomainName, apiVip)
```
Comment (eranco74) [inline internal/bminventory/inventory.go:3904]: No need to log apiVip. will be empty...
```suggestion
				log.WithError(err).Errorf("failed to find ip for single node installation")
```
